frontend/src/metabase/embedding/embedding-hub/hooks/use-completed-embedding-hub-steps.ts

```diff
@@ -1,67 +1,35 @@
 import { useMemo } from "react";
-import _ from "underscore";
 
-import { useGetPermissionsGraphQuery } from "metabase/api";
-import { useSetting } from "metabase/common/hooks";
+import { useGetEmbeddingHubChecklistQuery } from "metabase/api/embedding-hub";
 
 import type { EmbeddingHubStepId } from "../types";
 
 /**
  * Embedding Hub completion steps should be derived by the instance state at the time, or tracked manually in instance settings for some of them.
  */
-export const useCompletedEmbeddingHubSteps = (): Record<
-  EmbeddingHubStepId,
-  boolean
-> => {
-  const isJwtEnabled = useSetting("jwt-enabled");
-  const isSamlEnabled = useSetting("saml-enabled");
-  const isJwtConfigured = useSetting("jwt-configured");
-  const isSamlConfigured = useSetting("saml-configured");
-
-  const isSsoReady =
-    (isJwtEnabled && isJwtConfigured) || (isSamlEnabled && isSamlConfigured);
-
-  const { data: permissionsGraph } = useGetPermissionsGraphQuery();
-
-  const hasConfiguredSandboxes = useMemo(
-    () => getValuesFlat(permissionsGraph).includes("sandboxed"),
-    [permissionsGraph],
-  );
-
-  const isTestEmbedCreated = useSetting(
-    "embedding-hub-test-embed-snippet-created",
-  );
-
-  const isProductionEmbedCreated = useSetting(
-    "embedding-hub-production-embed-snippet-created",
-  );
-
-  return useMemo(() => {
-    return {
-      "create-test-embed": isTestEmbedCreated ?? false,
-      "add-data": false,
-      "create-dashboard": false,
-      "configure-row-column-security": hasConfiguredSandboxes,
-      "secure-embeds": isSsoReady,
-      "embed-production": isProductionEmbedCreated ?? false,
-    };
-  }, [
-    isTestEmbedCreated,
-    hasConfiguredSandboxes,
-    isSsoReady,
-    isProductionEmbedCreated,
-  ]);
+export const useCompletedEmbeddingHubSteps = (): {
+  data: Record<EmbeddingHubStepId, boolean>;
+  isLoading: boolean;
+} => {
+  const { data: embeddingHubChecklist, isLoading } =
+    useGetEmbeddingHubChecklistQuery(undefined, {
+      refetchOnMountOrArgChange: true,
```

Comment:
I suppose this re-fetches the checklist every time the page visited, which is good so we can keep the checklist updated after key actions 👍🏻